test/acceptance/features/environment.py

```diff
@@ -49,9 +55,28 @@ def before_all(_context):
         _context.sbo_namespace = output
     else:
         assert False, f"TEST_ACCEPTANCE_START_SBO={start_sbo} is currently unsupported."
+    _context.no_scenarios_failed = True
 
 
 def before_scenario(_context, _scenario):
     _context.bindings = dict()
     output, code = cmd.run(f'{ctx.cli} get ns default -o jsonpath="{{.metadata.name}}"')
     assert code == 0, f"Checking connection to OS cluster by getting the 'default' project failed: {output}"
+
+
+def after_scenario(_context, scenario):
+    namespace = os.getenv("TEST_NAMESPACE")
```

Comment:
The `TEST_NAMESPACE` env variable is not always the namespace in which the scenario is executed.

All scenarios use the `Namespace "name" is used` or `Namespace [ENV] is used` step in the background which sets the `context.namespace` attribute.

You should first check if that is set (by `if "namespace" in _context:`) and use that value, otherwise use the `TEST_NAMESPACE` env variable as the default.

Thx

Reply:
hi 
do u mean something like this

Reply:
if(os.getenv("TEST_NAMESPACE")):
        namespace = os.getenv("TEST_NAMESPACE")
    elif "namespace" in _context):
        namespace = ****** ->> to be filled
    else: 
        print(f"No namespace set in context skipping deletion")

Reply:
Rather something like:
```python
    if "namespace" in _context:
        namespace = _context.namespace.name
    elif os.getenv("TEST_NAMESPACE"):
        namespace = os.getenv("TEST_NAMESPACE")
    else:
        print(f"No namespace set in context nor TEST_NAMESPACE env variable is set, skipping deletion")
        return
```
The namespace in context should have precedence - after all it is supposed to be set by the "Namespace ... is used" step in the scenario, so it should be the one cleaned.

Reply:
FYI, the `_context.namespace` is not a `string` but an instance of the `Namespace` class ([`test/acceptance/features/steps/namespace.py`](https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/steps/namespace.py)) - See https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/steps/steps.py#L40